src/osd/OSDMap.cc

```diff
@@ -2068,7 +2068,8 @@ void OSDMap::_get_temp_osds(const pg_pool_t& pool, pg_t pg,
 
 void OSDMap::pg_to_raw_osds(pg_t pg, vector<int> *raw, int *primary) const
 {
-  *primary = -1;
```

Comment:
> the condition 'if(primary)' is redundant

@dalgaaf i checked all caller site of `OSDMap::pg_to_raw_osds()`, and it turns out it is redundant.

Reply:
I will then remove the redundant check

Reply:
actually I change the check to do the same as in the next function

Reply:
okay, it's still redundant. and if `primary` is `nullptr`, it crashes just like the next function.